source/MaterialXRender/Handlers/OiioImageLoader.cpp

```diff
@@ -53,31 +106,51 @@ bool OiioImageLoader::acquireImage(const FilePath& filePath,
     }
 
     OIIO::ImageSpec imageSpec = imageInput->spec();
-    if (imageSpec.format == OIIO::TypeDesc::HALF)
-    {
-        // Due to display issue with 16-bit tiled exrs,
-        // treat as 32-bit float.
-        imageSpec.set_format(OIIO::TypeDesc::FLOAT);
-    }
-
-    if (imageSpec.format != OIIO::TypeDesc::UINT8 &&
-        imageSpec.format != OIIO::TypeDesc::FLOAT)
+    switch (imageSpec.format.basetype)
     {
-        return false;
-    }
+        case OIIO::TypeDesc::UINT8:
+        {
+            imageDesc.baseType = ImageDesc::BaseType::UINT8;
+            break;
+        }
+        case OIIO::TypeDesc::FLOAT:
+        {
+            imageDesc.baseType = ImageDesc::BaseType::FLOAT;
+            break;
+        }
+        case OIIO::TypeDesc::HALF:
+        {
+            if (restrictions && restrictions->supportedBaseTypes.count(ImageDesc::BaseType::HALF_FLOAT) == 0)
+            {
+                // 16-bit float is not support so loadin as 32-bit float.
+                imageSpec.set_format(OIIO::TypeDesc::FLOAT);
+                imageDesc.baseType = ImageDesc::BaseType::FLOAT;
```

Comment:
Restrictions say we support HALF_FLOAT so why do we convert to FLOAT? (The logic is kind of inverted here)

There is no check that FLOAT is supported before using it.